lib/net_mosq.c

```diff
@@ -326,6 +326,9 @@ int net__try_connect(struct mosquitto *mosq, const char *host, uint16_t port, mo
 	}
 
 	for(rp = ainfo; rp != NULL; rp = rp->ai_next){
+		/* Only ignore ipv6 if there is another entry in the list */
+		if (mosq->ignore_ipv6 && rp->ai_family == PF_INET6 && NULL != rp->ai_next) continue;
```

Comment:
So now you're goign to try the _last_ ipv6 address if there's only ipv6 addresses, instead of the first one?  I don't really think that's what the resolver intends when it gives you a list.